x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EnrichExec.java

```diff
@@ -220,4 +221,14 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(super.hashCode(), mode, matchType, matchField, policyName, policyMatchField, concreteIndices, enrichFields);
     }
+
+    @Override
+    public ExecuteLocation executesOn() {
+        if (mode == Enrich.Mode.REMOTE) {
+            return ExecuteLocation.REMOTE;
+        } else if (mode == Enrich.Mode.COORDINATOR) {
+            return ExecuteLocation.COORDINATOR;
+        }
+        return ExecuteLocation.ANY;
```

Comment:
Nit:
```suggestion
        return switch (mode) {
            case REMOTE -> ExecuteLocation.REMOTE;
            case COORDINATOR -> ExecuteLocation.COORDINATOR;
            case ANY ->  ExecuteLocation.ANY;
        };
```

Reply:
Yes I guess it's finally time to start using those :)